x-pack/plugin/snapshot-repo-test-kit/src/main/java/org/elasticsearch/repositories/blobstore/testkit/BlobAnalyzeAction.java

```diff
@@ -178,7 +178,8 @@ protected void doExecute(Task task, Request request, ActionListener<Response> li
                 throw new IllegalArgumentException("repository [" + request.getRepositoryName() + "] is read-only");
             }
             final BlobStoreRepository blobStoreRepository = (BlobStoreRepository) repository;
-            final BlobContainer blobContainer = blobStoreRepository.blobStore().blobContainer(new BlobPath().add(request.blobPath));
+            final BlobPath path = blobStoreRepository.basePath().add(request.blobPath);
```

Comment:
We weren't using the provided base_path during the repo test kit analysis. I think this could lead to permission problems.

Reply:
Oof yes that's a good catch. Would you move these changes into a separate PR? I think we should strengthen some of the existing tests to check this too, for instance `RepositoryAnalysisSuccessIT`:

```diff
diff --git a/x-pack/plugin/snapshot-repo-test-kit/src/internalClusterTest/java/org/elasticsearch/repositories/blobstore/testkit/RepositoryAnalysisSuccessIT.java b/x-pack/plugin/snapshot-repo-test-kit/src/internalClusterTest/java/org/elasticsearch/repositories/blobstore/testkit/RepositoryAnalysisSuccessIT.java
index b6ddc8a7d59..f5d2c2b174d 100644
--- a/x-pack/plugin/snapshot-repo-test-kit/src/internalClusterTest/java/org/elasticsearch/repositories/blobstore/testkit/RepositoryAnalysisSuccessIT.java
+++ b/x-pack/plugin/snapshot-repo-test-kit/src/internalClusterTest/java/org/elasticsearch/repositories/blobstore/testkit/RepositoryAnalysisSuccessIT.java
@@ -17,6 +17,7 @@ import org.elasticsearch.common.blobstore.BlobStore;
 import org.elasticsearch.common.blobstore.DeleteResult;
 import org.elasticsearch.common.blobstore.support.PlainBlobMetadata;
 import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.common.unit.ByteSizeValue;
 import org.elasticsearch.common.unit.TimeValue;
 import org.elasticsearch.common.util.BigArrays;
@@ -49,6 +50,7 @@ import java.util.concurrent.atomic.AtomicReference;
 import java.util.function.Consumer;
 import java.util.stream.Collectors;

+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.lessThanOrEqualTo;
 import static org.hamcrest.Matchers.nullValue;
@@ -56,6 +58,8 @@ import static org.hamcrest.Matchers.startsWith;

 public class RepositoryAnalysisSuccessIT extends AbstractSnapshotIntegTestCase {

+    private static final String BASE_PATH_SETTING_KEY = "base_path";
+
     @Before
     public void suppressConsistencyChecks() {
         disableRepoConsistencyCheck("repository is not used for snapshots");
@@ -68,9 +72,17 @@ public class RepositoryAnalysisSuccessIT extends AbstractSnapshotIntegTestCase {

     public void testRepositoryAnalysis() {

-        createRepositoryNoVerify("test-repo", TestPlugin.ASSERTING_REPO_TYPE);
+        final Settings.Builder settings = Settings.builder();
+        if (randomBoolean()) {
+            settings.put(BASE_PATH_SETTING_KEY, randomAlphaOfLength(10));
+        }

-        final AssertingBlobStore blobStore = new AssertingBlobStore();
+        assertAcked(clusterAdmin().preparePutRepository("test-repo")
+                .setVerify(false)
+                .setType(TestPlugin.ASSERTING_REPO_TYPE)
+                .setSettings(settings));
+
+        final AssertingBlobStore blobStore = new AssertingBlobStore(settings.get(BASE_PATH_SETTING_KEY));
         for (final RepositoriesService repositoriesService : internalCluster().getInstances(RepositoriesService.class)) {
             try {
                 ((AssertingRepository) repositoriesService.repository("test-repo")).setBlobStore(blobStore);
@@ -133,10 +145,19 @@ public class RepositoryAnalysisSuccessIT extends AbstractSnapshotIntegTestCase {
                     clusterService,
                     bigArrays,
                     recoverySettings,
-                    new BlobPath()
+                    buildBlobPath(metadata.settings())
                 )
             );
         }
+
+        private static BlobPath buildBlobPath(Settings settings) {
+            final String base_path = settings.get(BASE_PATH_SETTING_KEY);
+            if (base_path == null) {
+                return BlobPath.cleanPath();
+            } else {
+                return BlobPath.cleanPath().add(base_path);
+            }
+        }
     }

     static class AssertingRepository extends BlobStoreRepository {
@@ -168,6 +189,8 @@ public class RepositoryAnalysisSuccessIT extends AbstractSnapshotIntegTestCase {

     static class AssertingBlobStore implements BlobStore {

+        private final String pathPrefix;
+
         @Nullable // if no current blob container
         private String currentPath;

@@ -179,9 +202,13 @@ public class RepositoryAnalysisSuccessIT extends AbstractSnapshotIntegTestCase {
         private long maxBlobSize = new RepositoryAnalyzeAction.Request("dummy").getMaxBlobSize().getBytes();
         private long maxTotalBlobSize = new RepositoryAnalyzeAction.Request("dummy").getMaxTotalDataSize().getBytes();

+        public AssertingBlobStore(@Nullable String basePath) {
+            this.pathPrefix = basePath == null ? "" : basePath + "/";
+        }
+
         @Override
         public BlobContainer blobContainer(BlobPath path) {
-            assertThat(path.buildAsString(), startsWith("temp-analysis-"));
+            assertThat(path.buildAsString(), startsWith(pathPrefix + "temp-analysis-"));

             synchronized (this) {
                 if (currentPath == null) {
```